src/InteractiveWindow/Editor/InteractiveWindow.cs

```diff
@@ -3067,116 +3013,53 @@ private static void DoEvents()
-            {
-                _window.HistoryPrevious(search);
-            }
-
-            public void HistorySearchNext()
-            {
-                _window.HistorySearchNext();
-            }
-
-            public void HistorySearchPrevious()
-            {
-                _window.HistorySearchPrevious();
-            }
-
-            public void Home(bool extendSelection)
-            {
-                _window.Home(extendSelection);
-            }
-
-            public bool Paste()
-            {
-                return _window.Paste();
-            }
-
-            public Task<ExecutionResult> ResetAsync(bool initialize = true)
-            {
-                return _window.ResetAsync(initialize);
-            }
-
-            public bool Return()
-            {
-                return _window.Return();
-            }
-
-            public void SelectAll()
-            {
-                _window.SelectAll();
-            }
-
-            public bool TrySubmitStandardInput()
-            {
-                return _window.TrySubmitStandardInput();
-            }
+        private enum State
+        {
+            /// <summary>
+            /// Initial state.  <see cref="IInteractiveWindow.InitializeAsync"/> hasn't been called.
+            /// Transition to <see cref="Initializing"/> when <see cref="IInteractiveWindow.InitializeAsync"/> is called.
+            /// Transition to <see cref="Resetting"/> when <see cref="ResetAsync"/> is called.
+            /// </summary>
+            Starting,
+            /// <summary>
+            /// In the process of calling <see cref="IInteractiveWindow.InitializeAsync"/>.
+            /// Transition to <see cref="WaitingForInput"/> when finished (in <see cref="ProcessPendingSubmissions"/>).
+            /// Transition to <see cref="Resetting"/> when <see cref="ResetAsync"/> is called.
+            /// </summary>
+            Initializing,
+            /// <summary>
+            /// In the process of calling <see cref="ResetAsync"/>.
+            /// Transition to <see cref="WaitingForInput"/> when finished (in <see cref="ProcessPendingSubmissions"/>).
+            /// Note: Should not see <see cref="ResetAsync"/> calls while in this state.
+            /// </summary>
+            Resetting,
+            /// <summary>
+            /// Prompt has been displayed - waiting for the user to make the next submission.
+            /// Transition to <see cref="ExecutingInput"/> when <see cref="IInteractiveWindowOperations.ExecuteInput"/> is called.
+            /// Transition to <see cref="Resetting"/> when <see cref="ResetAsync"/> is called.
+            /// </summary>
+            WaitingForInput,
+            /// <summary>
+            /// Executing the user's submission.
+            /// Transition to <see cref="WaitingForInput"/> when finished (in <see cref="ProcessPendingSubmissions"/>).
+            /// Transition to <see cref="Resetting"/> when <see cref="ResetAsync"/> is called.
```

Comment:
@ManishJayaswal Yes.  Presently, that's the only way to cancel a long-running computation.